### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-07-09

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/Congruence/Basic.lean [hunk 1]
```diff
@@ -195,11 +195,6 @@ theorem toSetoid_inj {c d : Con M} (H : c.toSetoid = d.toSetoid) : c = d :=
   ext <| ext_iff.1 H
 #align con.to_setoid_inj Con.toSetoid_inj
 #align add_con.to_setoid_inj AddCon.toSetoid_inj
-
-/-- Iff version of extensionality rule for congruence relations. -/
-@[to_additive "Iff version of extensionality rule for additive congruence relations."]
-theorem ext_iff {c d : Con M} : (∀ x y, c x y ↔ d x y) ↔ c = d :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/Congruence/Basic.lean:201]: Also could be flipped

File: Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean [hunk 2]
```diff
@@ -346,9 +346,9 @@ theorem ext {p q : AffineSubspace k P} (h : ∀ x, x ∈ p ↔ x ∈ q) : p = q
 #align affine_subspace.ext AffineSubspace.ext
 
 -- Porting note: removed `simp`, proof is `simp only [SetLike.ext'_iff]`
-theorem ext_iff (s₁ s₂ : AffineSubspace k P) : (s₁ : Set P) = s₂ ↔ s₁ = s₂ :=
+theorem ext_iff' (s₁ s₂ : AffineSubspace k P) : (s₁ : Set P) = s₂ ↔ s₁ = s₂ :=
   SetLike.ext'_iff.symm
```
Comment (Ruben-VandeVelde) [inline Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean:350]: Another one

File: Mathlib/NumberTheory/Bernoulli.lean [hunk 3]
```diff
@@ -341,7 +341,8 @@ theorem sum_range_pow (n p : ℕ) :
         mk fun p =>
           ∑ i ∈ range (p + 1), bernoulli i * (p + 1).choose i * (n : ℚ) ^ (p + 1 - i) / (p + 1)! by
       rw [← div_eq_iff (hne p), div_eq_mul_inv, sum_mul]
-      rw [PowerSeries.ext_iff] at this
+      -- Adaptation_note (2024-07-08): this was rw, but that created an instance goal
```
Comment (Ruben-VandeVelde) [inline Mathlib/NumberTheory/Bernoulli.lean:344]: We have the special syntax for adaptation notes now  (and I think the linter would have complained if not for the sneaky underscore)

File: Mathlib/RingTheory/DedekindDomain/Factorization.lean [hunk 4]
```diff
@@ -73,7 +73,7 @@ theorem Ideal.finite_factors {I : Ideal R} (hI : I ≠ 0) :
     Finite.of_injective (fun v => (⟨(v : HeightOneSpectrum R).asIdeal, v.2⟩ : { x // x ∣ I })) ?_
   intro v w hvw
   simp? at hvw says simp only [Subtype.mk.injEq] at hvw
-  exact Subtype.coe_injective ((HeightOneSpectrum.ext_iff (R := R) ↑v ↑w).mpr hvw)
+  exact Subtype.coe_injective (HeightOneSpectrum.ext hvw)
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/DedekindDomain/Factorization.lean:76]: .

**Review by Ruben-VandeVelde — APPROVED** [review]
Two comments I'd like you to fix before merging (Mathlib/CategoryTheory/Bicategory/Coherence.lean and Mathlib/NumberTheory/Bernoulli.lean).

I think we should try to keep the divergence between master and the bump branches minimal, so I'd like to see use make `ext_iff` `protected` on master, and fix the few lemmas that are currently in the wrong direction. That shouldn't block this PR, though, and I can try to make some time to work on that myself as well.

File: Mathlib/Data/Fin/Basic.lean [hunk 5]
```diff
@@ -841,8 +843,7 @@ lemma _root_.finCongr_eq_equivCast (h : n = m) : finCongr h = .cast (h ▸ rfl)
 
 @[simp]
 theorem cast_zero {n' : ℕ} [NeZero n] {h : n = n'} : cast h (0 : Fin n) =
-    by { haveI : NeZero n' := by {rw [← h]; infer_instance}; exact 0} :=
-  ext rfl
+    by { haveI : NeZero n' := by {rw [← h]; infer_instance}; exact 0} := rfl
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Data/Fin/Basic.lean:846]: Indeed, done.

File: Mathlib/Data/Finmap.lean [hunk 6]
```diff
@@ -163,9 +163,9 @@ theorem ext : ∀ {s t : Finmap β}, s.entries = t.entries → s = t
 #align finmap.ext Finmap.ext
 
 @[simp]
-theorem ext_iff {s t : Finmap β} : s.entries = t.entries ↔ s = t :=
-  ⟨ext, congr_arg _⟩
-#align finmap.ext_iff Finmap.ext_iff
+theorem ext_iff' {s t : Finmap β} : s.entries = t.entries ↔ s = t :=
+  Finmap.ext_iff.symm
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Data/Finmap.lean:167]: Ditto. :-)

File: Mathlib/Data/Sigma/Lex.lean [hunk 7]
```diff
@@ -116,7 +116,7 @@ instance [IsAsymm ι r] [∀ i, IsAntisymm (α i) (s i)] : IsAntisymm _ (Lex r s
     · exact (asymm hij hji).elim
     · exact (irrefl _ hij).elim
     · exact (irrefl _ hji).elim
-    · exact ext rfl (heq_of_eq <| antisymm hab hba)⟩
+    · exact congr_arg (Sigma.mk _ ·) <| antisymm hab hba⟩
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Data/Sigma/Lex.lean:119]: Done.

File: Mathlib/GroupTheory/Congruence/Basic.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/GroupTheory/Congruence/Basic.lean:201]: Again unsure what you intended here.

File: Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean (hunk 2 above)
Comment (semorrison) [reply Mathlib/LinearAlgebra/AffineSpace/AffineSubspace.lean:350]: As above, not sure what change on master you had in mind.

File: Mathlib/NumberTheory/Bernoulli.lean (hunk 3 above)
Comment (semorrison) [reply Mathlib/NumberTheory/Bernoulli.lean:344]: Fixed.

File: Mathlib/RingTheory/DedekindDomain/Factorization.lean (hunk 4 above)
Comment (semorrison) [reply Mathlib/RingTheory/DedekindDomain/Factorization.lean:76]: Backported.

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
bors p=10

**Pull request closed without merging**